### ISSUE TITLE
Fix deprecation notices on PHP nightly

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#1555]
## Bug Description

Function `ReflectionParameter::isArray()` is deprecated, see e.g. https://travis-ci.com/github/google/site-kit-wp/jobs/333190714

For background information on how to replace usage, see https://github.com/php/php-src/pull/5209 (`ReflectionParameter::getType()`). We may need to keep both and use them depending on PHP version active.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be no PHP warnings generated by running our tests.
* Travis-CI build for PHP nightly and WordPress trunk should pass.

## Implementation Brief

* 

## QA Brief

* 

## Changelog entry

*

**Comment by rahul2106** [comment]
Please assign the issue to me. Interested in starting my first contribution here.

**Comment by aaemnnosttv** [comment]
Thanks for your enthusiasm @rahul2106. Unfortunately this issue is still being defined so it's not quite ready for execution yet.

**Comment by aaemnnosttv** [comment]
@felixarntz not sure there is anything we can do about this right now. I tried upgrading PHPUnit to the latest supported version (v7) and it still raises deprecated warnings. See #1603 

I checked WordPress core travis configuration and the PHP8 job is added to the allowed failures, and currently failing.
- https://travis-ci.com/github/WordPress/wordpress-develop/builds/168543704
- https://travis-ci.com/github/WordPress/wordpress-develop/jobs/340875516#L2945

Moving this back to AC for now, but it's safe to say that this one isn't actionable right now.
